### ISSUE TITLE
RNFFprobe.getMediaInformation is blocked while RNFFmpeg.execute is running - only on Android

### ISSUE THREAD

**Issue by Rotemy** [tanersener/react-native-ffmpeg#188]
Let me know if you need more info.

**Comment by tanersener** [comment]
See #62

**Comment by Rotemy** [comment]
Hi @tanersener sorry for not providing template and thank you for maintaining this awesome library.

I managed to fixed the issue by replacing `asyncTask.execute();` with `asyncTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);`

I will patch the patch I did shortly so other people can use it.

**Comment by Rotemy** [comment]
Here is the patch

```
diff --git a/node_modules/react-native-ffmpeg/android/src/main/java/com/arthenica/reactnative/RNFFmpegModule.java b/node_modules/react-native-ffmpeg/android/src/main/java/com/arthenica/reactnative/RNFFmpegModule.java
index 0824013..25f4dbc 100644
--- a/node_modules/react-native-ffmpeg/android/src/main/java/com/arthenica/reactnative/RNFFmpegModule.java
+++ b/node_modules/react-native-ffmpeg/android/src/main/java/com/arthenica/reactnative/RNFFmpegModule.java
@@ -52,6 +52,7 @@ import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import android.os.AsyncTask;
 
 public class RNFFmpegModule extends ReactContextBaseJavaModule {
 
@@ -108,13 +109,13 @@ public class RNFFmpegModule extends ReactContextBaseJavaModule {
     @ReactMethod
     public void executeFFmpegWithArguments(final ReadableArray readableArray, final Promise promise) {
         final RNFFmpegExecuteFFmpegAsyncArgumentsTask asyncTask = new RNFFmpegExecuteFFmpegAsyncArgumentsTask(promise, readableArray);
-        asyncTask.execute();
+        asyncTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
     }
 
     @ReactMethod
     public void executeFFprobeWithArguments(final ReadableArray readableArray, final Promise promise) {
         final RNFFmpegExecuteFFprobeAsyncArgumentsTask asyncTask = new RNFFmpegExecuteFFprobeAsyncArgumentsTask(promise, readableArray);
-        asyncTask.execute();
+        asyncTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
     }
 
     @ReactMethod
@@ -225,7 +226,7 @@ public class RNFFmpegModule extends ReactContextBaseJavaModule {
     @ReactMethod
     public void getMediaInformation(final String path, final Promise promise) {
         final RNFFmpegGetMediaInformationAsyncTask asyncTask = new RNFFmpegGetMediaInformationAsyncTask(path, promise);
-        asyncTask.execute();
+        asyncTask.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
     }
 
     @ReactMethod
```

**Comment by tanersener** [comment]
Thanks, I need to look into the details of this issue before making any changes. I'll update the issue after understanding the reason.

**Comment by Rotemy** [comment]
@tanersener Let me know if you need anything from me

**Comment by tanersener** [comment]
`execute()` method uses `AsyncTask.SERIAL_EXECUTOR`, which allows only one task at a time. This was the reason why your `getMediaInformation()` call was blocked. 

Using `AsyncTask.THREAD_POOL_EXECUTOR` fixes this problem. Thanks for reporting this issue and also for providing the patch. It is fixed in the `development` branch.

**Comment by Rotemy** [comment]
Thank you again for this amazing library.

**Comment by tanersener** [comment]
Fixed in `v0.5.0`.
